### PR TITLE
respect --if-match and --if-none-match at getObject()

### PR DESCRIPTION
**Pull request by lre**
## Description
When calling getObject there are 2 hedder parameters that can be set
to test whether the e-tag exists and should or should not match.

## Tasks
- [x] I have signed the [CLA](http://adobe.github.io/cla.html).
- [x] I have written tests and verified that they fail without my change.

### CODE REVIEW

**Comment by timoe** [comment]
Thanks for this PR @lre, I'll take a look.

File: server/src/main/java/com/adobe/testing/s3mock/FileStoreController.java [hunk 1]
```diff
@@ -470,6 +475,8 @@ public CopyObjectResult copyObject(@PathVariable final String destinationBucket,
       produces = "application/x-www-form-urlencoded")
   public void getObject(@PathVariable final String bucketName,
       @RequestHeader(value = RANGE, required = false) final Range range,
+      @RequestHeader(value = IFMATCH, required = false) final List<String> match,
```
Comment (timoe) [inline server/src/main/java/com/adobe/testing/s3mock/FileStoreController.java]: please use std. Headers from `org.springframework.http.HttpHeaders`. `BetterHeaders` is meant to wrap AWS API specific headers.
Same applies to `IFNONEMATCH` Header

File: server/src/main/java/com/adobe/testing/s3mock/util/BetterHeaders.java [hunk 2]
```diff
@@ -30,6 +30,9 @@
   public static final String NOT_SERVER_SIDE_ENCRYPTION = NOT + SERVER_SIDE_ENCRYPTION;
 
   public static final String RANGE = "Range";
+  public static final String IFMATCH = "if-match";
```
Comment (timoe) [inline server/src/main/java/com/adobe/testing/s3mock/util/BetterHeaders.java]: See comment on FileStoreController

File: server/src/test/java/com/adobe/testing/s3mock/its/AmazonClientUploadIT.java [hunk 3]
```diff
@@ -707,4 +707,64 @@ public void shouldAddAndRetrieveTags() {
...
+    final S3Object s3ObjectWithEtag = s3Client.getObject(requestWithEtag);
+    final S3Object s3ObjectWithHoutEtag = s3Client.getObject(requestWithHoutEtag);
+    final String s3ObjectWithEtagDownloadedHash = HashUtil
+            .getDigest(s3ObjectWithEtag.getObjectContent());
+    final String s3ObjectWithHoutEtagDownloadedHash = HashUtil
+            .getDigest(s3ObjectWithHoutEtag.getObjectContent());
+    final InputStream uploadFileIs = new FileInputStream(uploadFile);
+    final String uploadHash = HashUtil.getDigest(uploadFileIs);
+
+    assertThat("", uploadHash, is(equalTo(s3ObjectWithEtagDownloadedHash)));
```
Comment (timoe) [inline server/src/test/java/com/adobe/testing/s3mock/its/AmazonClientUploadIT.java]: please add a suitable assertion message here and for other uses of `assertThat` in this test

File: server/src/test/java/com/adobe/testing/s3mock/its/AmazonClientUploadIT.java [hunk 4]
```diff
@@ -707,4 +707,64 @@ public void shouldAddAndRetrieveTags() {
...
+
+    s3Client.createBucket(BUCKET_NAME);
+    PutObjectResult returnObj = s3Client.putObject(
+            new PutObjectRequest(BUCKET_NAME,
+                    uploadFile.getName(),
+                    uploadFile));
+
+    // wit eTag
+    GetObjectRequest requestWithEtag = new GetObjectRequest(BUCKET_NAME, uploadFile.getName());
+    requestWithEtag.setMatchingETagConstraints(new ArrayList<String>() { {
```
Comment (timoe) [inline server/src/test/java/com/adobe/testing/s3mock/its/AmazonClientUploadIT.java]: please use `java.util.Collections#singletonList(...)` here. Same for other `setNonmatchingETagConstraints` resp. `setMatchingETagConstraints` invocations

**Comment by timoe** [comment]
Hi @lre,

any updates here? Would you like to update this PR? I'm going to merge it afterwards.

Thanks

**Comment by lre** [comment]
Hi @timoe,

I have added my changes, just had to get the CLA figured out 😄 

Thanks

**Comment by timoe** [comment]
Thanks for your update @lre. PR got merged, I'll do a release

**Comment by timoe** [comment]
@lre: release done

**Pull request merged**